README
===============================================================================
                             Blackjack
===============================================================================

Blackjack was dropped from Gnome-games after version 2.28.2. Gnome-games is a
collection of simple, but addictive, games from the GNOME desktop project.

Blackjack is a casino card game also known as 21. There is no use of real money
in this version of the game.

License
-------

This is Free Software and there are no restrictions on your use and
enjoyment of it. In most countries copyright law would prevent you
from sharing these programs, either as is or in a modified
form. However we provide a license to allow you to modify and
redistribute this code. This is the GNU General Public License and it
can be found in the file COPYING. This allows you to do almost
anything you would like with this software, including selling
it. There are only two catches, firstly you can't restrict anyone else
from copying the code, you have to give them the same freedoms you
received (including access to the source code). Secondly, because the
code may have been modified and we have no idea what state it will be
in when it reaches you, we cannot provide a warranty. All the details
can be found in the file COPYING which should have accompanied this
software. If it didn't you can find a copy at

http://www.gnu.org/copyleft/gpl.html

or write to the Free Software Foundation, Inc., 59 Temple Place, Suite
330, Boston, MA 02111-1307 USA

Some parts of this software, notably some of the artwork, are licensed
under different terms or have been placed in the public domain. None
of these terms are more restrictive than what has been described
above. Please check the artwork files for full details. If there is no
information embedded within the file or alongside it then it is licensed
under the GPL.

Contributing
------------

See the file HACKING for details about coding style.

Requirements
------------

This is the list of required software packages for gnome-games:

 - GNOME 2.16.0
 - GTK+ 2.8.0
 - GNOME Doc Utils - gnome-doc-utils 0.10.0
 - Cairo 1.0
 - RSVG 2.14.0
 - libxml 2.4.0
 - scrollkeeper
 - expat

Building Blackjack
--------------------


NEWS
gnome-games 2.29.1
==================

To the users of Blackjack, we are sorry to inform you that the game has been
dropped from GNOME Games.

gnome-games 2.28.2
==================

gnome-games 2.28.1
==================

gnome-games 2.28.0
==================

gnome-games 2.27.92
===================

gnome-games 2.27.90
===================

gnome-games 2.27.5
==================

Tim's branch was merged.

gnome-games 2.27.4
==================

Mostly just bugfixes. New features include better maemo/win32 support.

gnome-games 2.27.3
==================

No major changes, just bug fixes all over the place.

Christian Persch (32):
      Don't build games-help on hildon yet
      Only include games-preimage.h if HAVE_RSVG
      GTK_STOCK_FULLSCREEN only exists on gtk >= 2.8
      Fix the build on hildon
      Use g_mkdir_with_parents and mode 0700 to create the config dir

Claude Paroz (3):
      Fix GNOME feedback link in docs

Jason D. Clinton (3):
      Bump release version to 2.27.3

gnome-games 2.27.2
=================
Shortlog follows:

Christian Persch (3):
      libgames-support: remove leftover HAVE_GSTREAMER
      Initialise the canberra_gtk context only once

Jason D. Clinton (16):
      Post-release bump to 2.27.2

gnome-games 2.27.1
==================

General:
 - Don't append ? on ghelp URI with NULL section (Christian Persch)
 - Add Pulse Support to Docs (Paul Cutler)
 - Use g_timeout_add_seconds when available (Christian Persch)

gnome-games 2.26.0
==================

This release is the first stable release in the Gnome 2.26 desktop
module.

GNOME Games Extra Data 2.26.0
-----------------------------
* Add new card theme "Tango" created by Frederik Elwert

gnome-games 2.25.92
==================

gnome-games 2.25.91
==================

This second beta release brings many bugfixes and also many
code cleanups all over the place. Bugs from bugbuddy are now sent to a new
component 'BugBuddyBugs' to help manage the bug flow.

General:
  - Send bug from bugbuddy to new component 'BugBuddyBugs' (Jason Clinton,
    Thomas H.P. Andersen)
  - Rewrite of GamesFrame (Christian Persch)
  - Code cleanups (Christian Persch, Thomas H.P. Andersen)


gnome-games 2.25.90
==================

This is the first beta release of gnome-games.

gnome-games 2.25.5
==================

The gnome-games team is proud to release another version with lots of nice
improvements. We hope you'll love it :)

General:
  - Move GThread and gettext initialisation into games_runtime_init() (Christian
    Persch)
  - Refactored card themes code (Christian Persch)
  - Fixed typoed XML files (Jason Clinton)
  - Allow ChangeLog from git-svn (Jason Clinton)


gnome-games 2.25.4
==================

gnome-games 2.25.3
==================

gnome-games 2.25.2
==================

Lots of exciting new things in this release.


General:
  - Code cleanups and warnings (Robert Ancell, Christian Persch, Thomas Andersen,
    Zhang Sen)
  - Use games-runtime to locate files (Christian Persch)

Blackjack:
  - window-state-event callback needs to return gboolean, not void.
    (Christian Persch)

gnome-games 2.25.1
==================

This is the first 2.26 development release.

A huge amount of kudos should go out to Thomas
for his work on killing off libgnome.

All games:
  - GTK req. bumped to 2.14.0 for libgnome killoff (thomashpa, chpe)
  - Remove directory defines from CPPFLAGs (chpe)
  - Kill off libgnome usage (thomashpa, chpe) YAY! All games now start faster!
  - Whitespace cleanups
  - All games now build with gtk-only. libxml now checked for independently.
    (chpe)
  - -Wformat-security fixups (chpe)
  - -Werror fixups (chpe)
  - games-runtime: add runtime directory provider (thomashpa)
  - Refactor accel map loading code to load the accel map from
    .config/gnome-games/${game}.accels even in the HAVE_GNOME case, if the game
    doesn't use GnomeProgram (e.g. aisleriot). (chpe)

Blackjack:
  - Port game rules to GKeyFile

gnome-games 2.24.0
==================

This is the Gnome 2.24 stable release.

gnome-games 2.23.92
==================

If you meet one of the guys who contributed to this release please
be sure to thank the for being awesome :)

Blackjack:
  - Fix language in help (Claude Paroz)

gnome-games 2.23.91
==================

gnome-games 2.23.90
==================

Another development release of gnome-games leading up to the 2.24 release.

gnome-games 2.23.6
==================

Blackjack:
 - Revert bogus patch from bug #444806 (Christian Persch)

gnome-games 2.23.5
==================

This is the first release of Gnome Games that generates its ChangeLog from SVN
commit messages.

gnome-games 2.23.4
==================

gnome-games 2.23.3
==================

No new features over the previous 2.23.1 have been added.

gnome-games 2.23.1
==================

This is the first development release of the 2.23 series.

Overall:
  - define GAMES_STOCK_FULLSCREEN on hildon (Christian Persch)

gnome-games 2.22.0
==================

This is the final release of Gnome Games for 2.22.

gnome-games 2.21.92
==================
This is the release candidate for 2.22 of gnome-games.

gnome-games 2.21.91
==================
A fairly boring release, but still lots of fixes to polish off the games for the
big release.

General:
  - Update copyright year for 2008 (Andreas Røsdal)

gnome-games 2.21.90
==================
This is a pretty calm release of the development series as we prepare for an
awesome 2.22 soon.

Games support lib:
  - Use the same code to register stock icons and named icons. Add LTR and RTL
    variants for the GAMES_STOCK_UNDO/REDO_MOVE stock IDs (Christian Persch)
  - Rework stock icon registration in preparation for using the gtk facilities.
    Bug #511332 (Christian Persch)


gnome-games 2.21.5
==================
Anther rocking release from the gnome-games team.
Lots of fixes all over the place. Huge thanks goes to
the translators and documentors for the many fixes there also!

General:

src/menu.cpp
// -*- mode:C++; tab-width:8; c-basic-offset:8; indent-tabs-mode:nil -*-
/*
 * Blackjack - menu.cpp
 *
 * Copyright (C) 2003-2004 William Jon McCann <mccann@jhu.edu>
 *
 * This game is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2, or (at your option)
 * any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307
 * USA
 */

#include <config.h>

#include <string.h>
#include <stdlib.h>
#include <glib/gi18n.h>
#include <gtk/gtk.h>

using namespace std;

#include "games-help.h"
#include "games-stock.h"

#include "blackjack.h"
#include "menu.h"
#include "dialog.h"
#include "game.h"
#include "hand.h"

void
on_game_restart_activate (GtkAction *action,
                          gpointer   data)
{
        bj_game_new (bj_game_get_rules_file (), &seed);
}

void
on_game_new_activate (GtkAction *action,
                      gpointer   data)
{
        bj_game_new (bj_game_get_rules_file (), NULL);
}

void
on_game_hint_activate (GtkAction *action,
                       gpointer   data)
{
        show_hint_dialog ();
}

void
on_game_quit_activate (GtkAction *action,
                       gpointer   data)
{
        bj_quit_app ();
}

void
on_toolbar_activate (GtkToggleAction *action,
                     gpointer         data)
{
        gboolean value = gtk_toggle_action_get_active (action);
        bj_set_show_toolbar (value);
}

void
on_preferences_activate (GtkAction *action,
                         gpointer   data)
{
        show_preferences_dialog ();
}

void
on_control_deal_activate (GtkAction *action,
                         gpointer   data)
{
        bj_hand_new_deal ();
}

void
on_control_hit_activate (GtkAction *action,
                         gpointer   data)
{
        bj_hand_hit_with_delay ();
}

void
on_control_stand_activate (GtkAction *action,
                           gpointer   data)
{
        bj_hand_stand ();
}

void
on_control_surrender_activate (GtkAction *action,
                               gpointer   data)
{
        bj_hand_surrender ();
}

void
on_control_split_activate (GtkAction *action,
                               gpointer   data)
{
        bj_hand_split ();
}

void
on_control_double_activate (GtkAction *action,
                            gpointer   data)
{
        bj_hand_double ();
}

void
on_help_contents_activate (GtkAction *action,
                           gpointer   data)
{
  games_help_display(toplevel_window, "blackjack", NULL);
}

void
on_help_about_activate (GtkAction *action,
                        gpointer   data)
{
        const gchar *authors[] = {
                "William Jon McCann <mccann@jhu.edu>",
                "Eric Farmer <erfarmer@comcast.net>",
                NULL
        };

        const gchar *documenters[] = {
                "William Jon McCann <mccann@jhu.edu>",
                NULL
        };
	gchar *license = games_get_license (_("Blackjack"));

        gtk_show_about_dialog (GTK_WINDOW (toplevel_window),
                               "authors", authors,
                               "documenters", documenters,
                               "version", VERSION,
                               "copyright", "Copyright \xc2\xa9 2003-2008 William Jon McCann, Eric Farmer",
                               "comments", _("Blackjack is a casino-style card game.\n\n"
			       "Blackjack is a part of GNOME Games."),
                               "logo-icon-name", "gnome-blackjack",
                               "license", license,
			       "website-label", _("GNOME Games web site"),
                               "translator-credits", _("translator-credits"),
                               "website", "http://www.gnome.org/projects/gnome-games/",
                               "wrap-license", TRUE,
                               NULL);
	g_free (license);

        return;
}
